File: Programmers/L2_행렬테두리회전하기.js
function solution(rows, columns, queries) {
  let answer = [];

  let board = new Array(rows);
  for(let i = 0; i < rows; i++) {
    board[i] = new Array(columns);
    for(let j = 0; j < columns; j++) {
      board[i][j] = i*columns + (j+1)
    }
  }
  // const a = [...Array(rows)].map((_, r)=>[...Array(columns)].map((_, c)=>r*columns+c+1))

  queries.forEach((query) => {
    const [x1, y1, x2, y2] = query.map(q => q - 1)
    stack = [board[x1][y1]]

    for (let e = y1 + 1; e < y2 + 1; e++) {
      stack.push(board[x1][e])
      board[x1][e] = stack[stack.length-2]
    }

    for (let s = x1 + 1; s < x2 + 1; s++) {
      stack.push(board[s][y2])
      board[s][y2] = stack[stack.length-2]
    }

    for (let w = y2 - 1; w > y1 - 1; w--) {
      stack.push(board[x2][w])
      board[x2][w] = stack[stack.length-2]
    }

    for (let n = x2 - 1; n > x1 - 1; n--) {
      stack.push(board[n][y1])
      board[n][y1] = stack[stack.length-2]
    }
    answer.push(Math.min(...stack))
  })

  return answer;
}

File: Programmers/L2_행렬테두리회전하기.py
def solution(rows, columns, queries):
  answer = []
  board = [[(r*columns+c) for c in range(1, columns + 1)] for r in range(rows)]

  for x1, y1, x2, y2 in queries:
    x1, y1, x2, y2 = x1-1, y1-1, x2-1, y2-1
    stack = [board[x1][y1]]

    for e in range(y1+1, y2+1):
      stack.append(board[x1][e])
      board[x1][e] = stack[-2]

    for s in range(x1+1, x2+1):
      stack.append(board[s][y2])
      board[s][y2] = stack[-2]

    for w in range(y2-1, y1-1, -1):
      stack.append(board[x2][w])
      board[x2][w] = stack[-2]

    for n in range(x2-1, x1-1, -1):
      stack.append(board[n][y1])
      board[n][y1] = stack[-2]

    answer.append(min(stack))

  return answer